test/e2e/info_api_test.go

```diff
@@ -0,0 +1,178 @@
+	// '--endpoint' flag works properly works together with other flags ('--target', '--metadata' etc.).
+	// Register 2 sidecars and 1 storeGW using '--endpoint'.
+	// Register `sidecar3` twice to verify it is deduplicated.
+	q, err := e2ethanos.NewQuerierBuilder(s.SharedDir(), "1", []string{sidecar1.GRPCNetworkEndpoint()}).
+		WithTargetAddresses([]string{sidecar1.GRPCNetworkEndpoint()}).
+		WithMetadataAddresses([]string{sidecar1.GRPCNetworkEndpoint()}).
+		WithExemplarAddresses([]string{sidecar1.GRPCNetworkEndpoint()}).
+		WithRuleAddresses([]string{sidecar1.GRPCNetworkEndpoint()}).
+		WithEndpoints([]string{
+			sidecar1.GRPCNetworkEndpoint(),
+			sidecar2.GRPCNetworkEndpoint(),
+			sidecar3.GRPCNetworkEndpoint(),
+			sidecar3.GRPCNetworkEndpoint(),
+			str.GRPCNetworkEndpoint(),
+		}).
+		Build()
+	testutil.Ok(t, err)
+	testutil.Ok(t, s.StartAndWaitReady(q))
+
+	expected := map[string][]query.StoreStatus{
+		"sidecar": {
+			{
+				Name: "e2e_test_info-sidecar-alone1:9091",
+				LabelSets: []labels.Labels{{
+					{
+						Name:  "prometheus",
+						Value: "prom-alone1",
+					},
+					{
+						Name:  "replica",
+						Value: "0",
+					},
+				}},
+			},
+			{
+				Name: "e2e_test_info-sidecar-alone2:9091",
+				LabelSets: []labels.Labels{{
+					{
+						Name:  "prometheus",
+						Value: "prom-alone2",
+					},
+					{
+						Name:  "replica",
+						Value: "0",
+					},
+				}},
+			},
+			{
+				Name: "e2e_test_info-sidecar-alone3:9091",
+				LabelSets: []labels.Labels{{
+					{
+						Name:  "prometheus",
+						Value: "prom-alone3",
+					},
+					{
+						Name:  "replica",
+						Value: "0",
+					},
+				}},
+			},
+		},
+		"store": {
+			{
+				Name:      "e2e_test_info-store-gw-1:9091",
+				LabelSets: []labels.Labels{},
+			},
+		},
+	}
+
+	url := "http://" + path.Join(q.HTTPEndpoint(), "/api/v1/stores")
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	defer cancel()
+
+	err = runutil.Retry(time.Second, ctx.Done(), func() error {
+
+		resp, err := http.Get(url)
+		testutil.Ok(t, err)
+
+		body, err := ioutil.ReadAll(resp.Body)
+		testutil.Ok(t, err)
+
+		var res struct {
+			Data map[string][]query.StoreStatus `json:"data"`
+		}
+
+		if err = json.Unmarshal(body, &res); err != nil {
+			t.Fatalf("Error unmarshaling JSON body: %s", err)
+		}
+
+		if err = assertStoreStatus(t, "sidecar", res.Data, expected); err != nil {
+			return err
+		}
+
+		if err = assertStoreStatus(t, "store", res.Data, expected); err != nil {
+			return err
+		}
+
+		return nil
+	})
+	testutil.Ok(t, err)
+}
+
+func assertStoreStatus(t *testing.T, component string, res map[string][]query.StoreStatus, expected map[string][]query.StoreStatus) error {
+	t.Helper()
+
+	if len(res[component]) != len(expected[component]) {
+		return fmt.Errorf("Expected %d %s, got: %d", len(expected[component]), component, len(res[component]))
```

Comment:
Errors typically begin with a lower-case letter so that they would look nice when wrapped
```suggestion
		return fmt.Errorf("expected %d %s, got: %d", len(expected[component]), component, len(res[component]))
```